src/main/java/tutoraid/model/student/ProgressList.java

```diff
@@ -0,0 +1,152 @@
+package tutoraid.model.student;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a student's list of progress in TutorAid.
+ */
+public class ProgressList {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "ProgressList constructor either takes in no argument "
+            + "or takes in a String ArrayList of size no more than 10";
+
+    public final ArrayList<Progress> progressList;
+
+    /**
+     * Constructs a {@code ProgressList}.
+     */
+    public ProgressList() {
+        progressList = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a {@code ProgressList}.
+     *
+     * @param progressListInStringArrayList an arraylist of progress descriptions of the student
+     */
+    public ProgressList(ArrayList<String> progressListInStringArrayList) {
+        requireNonNull(progressListInStringArrayList);
+
+        this.progressList = new ArrayList<>();
+
+        if (!isValidProgressList(progressListInStringArrayList)) {
+            return;
+        }
+
+        for (int i = 0; i < progressListInStringArrayList.size(); i++) {
+            String currentProgressDescription = progressListInStringArrayList.get(i);
+            Progress currentProgress = new Progress(currentProgressDescription);
+
+            if (!currentProgress.isEmptyProgress()) {
+                this.addProgress(currentProgress);
+            }
+        }
+    }
+
+    /**
+     * Checks if a given string ArrayList is a valid list of progress descriptions.
+     *
+     * @param progressListInStringArrayList an ArrayList of strings
+     * @return true if all elements are valid progress description, false otherwise
+     */
+    public static boolean isValidProgressList(ArrayList<String> progressListInStringArrayList) {
+        if (progressListInStringArrayList.size() > 10) {
+            return false;
+        }
+
+        for (int i = 0; i < progressListInStringArrayList.size(); i++) {
+            if (progressListInStringArrayList.get(i) == null) {
+                return false;
+            }
+        }
+
+        for (int j = 0; j < progressListInStringArrayList.size(); j++) {
+            if (!Progress.isValidProgress(progressListInStringArrayList.get(j))) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Adds a new progress.
+     *
+     * @param progressToAdd the progress to be added
+     */
+    public void addProgress(Progress progressToAdd) {
+        requireNonNull(progressToAdd);
+        while (progressList.size() > 10) {
+            progressList.remove(0);
+        }
+
+        if (progressList.size() == 10) {
```

Comment:
Maybe can simplify this if-else block by just having `progress.remove(0)` inside the if block, completely removing the else block and just having `progressList.add(progressToAdd)` outside the if block